src/System.Private.ServiceModel/tests/Scenarios/Contract/Fault/FaultExceptionTests.4.0.0.cs

```diff
@@ -299,4 +301,74 @@ public static void FaultException_FaultContractAndKnownType_Echo()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+#if FULLXUNIT_NOTSUPPORTED
+    [Fact]
+#endif
+    [WcfFact]
+    [OuterLoop]
+    public static void TestAccessFaultContractInfo()
+    {
+        MyClientBase<IWcfService> client = null;
+        IWcfService serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            client = new MyClientBase<IWcfService>(customBinding, new EndpointAddress(Endpoints.DefaultCustomHttp_Address));
+            client.Endpoint.EndpointBehaviors.Add(new TestFaultContractInfosBehavior());
+            serviceProxy = client.ChannelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            var input = new object[] { new FaultDetail(), new KnownTypeA() };
+            // Use exiting service as we only need to verify FaultContractInfo can be accessed from ClientOperation
+            serviceProxy.TestFaultWithKnownType(null, input);
```

Comment:
I haven't figured out how to access the ClientOperations outside of the Behavior.